src/api/ans.ts

```diff
@@ -161,4 +165,51 @@ export class ANS {
   async renewDomain(args: { sender: Account; name: string; years?: 1 }): Promise<InputSingleSignerTransaction> {
     return renewDomain({ aptosConfig: this.config, ...args });
   }
+
+  /**
+   * Fetches a single name from the indexer
+   * @param args.name - A string of the name to retrieve, e.g. "test.aptos.apt"
+   * or "test.apt" or "test". Can be inclusive or exclusive of the .apt suffix.
+   * Can be a subdomain.
+   *
+   * @returns A promise of an ANSName or undefined
+   */
+  async getName(args: { name: string }): Promise<ANSName | undefined> {
+    return getName({ aptosConfig: this.config, ...args });
+  }
+
+  /**
+   * Fetches multiple names from the indexer. Can be used to fetch names owned
+   * by an address or subdomains of a domain.
+   *
+   * ```ts
+   * // Fetch all names owned by alice
+   * aptos.getNames({ query: 'owner', ownerAddress: alice.address });
+   *
+   * // Fetch all top level domains (excluding subdomains) owned by alice
+   * aptos.getNames({ query: 'owner:domains', ownerAddress: alice.address });
```

Comment:
This happens automatically at the contract level when: renewing a domain, setting a target address, and setting a primary name. Nothing we can do to seperate it.

Reply:
hmm I guess we can still have something like. I am just thinking on the dev exp and how it can be simplified
```
aptos.getAccountAllOwnedNames(ownerAddress)
aptos.getAccountOwnedDomains(ownerAddress)
aptos.getAccountOwnedSubdomains(ownerAddress)
aptos.getSubdomainForDomain(domain)
```

Reply:
Ohh I see. I thought you meant separate out the token migration from the base query. Sure, I can make seperate functions for each of these queries!